src/console_cmds.cpp

```diff
@@ -1049,6 +1049,54 @@ DEF_CONSOLE_CMD(ConNewGame)
 	return true;
 }
 
+DEF_CONSOLE_CMD(ConNewHeightMapGame)
+{
+	bool heightmap_loaded = true;
+
+	if (argc == 0) {
+		IConsoleHelp("Load a game by name or index. Usage: 'newheightmapgame [<file | number> [seed]]'");
+		IConsoleHelp("The server can force a new game using 'newheightmapgame'; any client joined will rejoin after the server is done generating the new game.");
+	}
+
+	if (argc >=1 && argc <= 3) {
+		const char *file = NULL;
+
+		if (argc == 1) {
+			file = _file_to_saveload.title;
+		}
+
+		if (argc >= 2) {
+			file = argv[1];
+		}
+
+		/* ConLoad command actually uses
+		_console_file_list.ValidateFileList()
+		here, but for this procedure build a file list for mere savegames, instead of merely validating */
+		_console_file_list.BuildFileList(FT_HEIGHTMAP, SLO_LOAD);
+		const FiosItem *item = _console_file_list.FindItem(file);
+		if (item != NULL) {
+			if (GetAbstractFileType(item->type) == FT_HEIGHTMAP) {
+				_switch_mode = SM_LOAD_HEIGHTMAP;
+				_file_to_saveload.SetMode(item->type);
+				_file_to_saveload.SetName(FiosBrowseTo(item));
+				_file_to_saveload.SetTitle(item->title);
+
+				StartNewHeightMapGameWithoutGUI((argc == 3) ? strtoul(argv[2], NULL, 10) : GENERATE_NEW_SEED);
+			} else {
+				IConsolePrintF(CC_ERROR, "%s: Not a heightmap.", file);
+			}
+		} else {
+			IConsolePrintF(CC_ERROR, "%s: No such file or directory.", file);
+		}
+
+		_console_file_list.InvalidateFileList();
+	}
+
+	if (argc > 3) heightmap_loaded = false;
+
+	return heightmap_loaded;
```

Comment:
If argc is 0 then heightmap_loaded will be true, although it didn't do it?

Reply:
Look at `ConNewGame`, it does exactly the same.

Are `IConsoleHelp` calls are enough? Is returning true without doing anything else is *not* bad?
I am not enough of a pro to decide what should, rather than what is, thus I mimic what is existing.

Reply:
You should always early return after handling the help case at the top of the function, to make it super obvious that requesting help has no side effects.

Reply:
I believe the preferred style is to use a shallow control flow with early returns to reject invalid parameters, collect the valid parameters into a set of variables, and then execute the command on the top level of the function.

Reply:
@berbe ConNewGame has a `return true;` inside the help block. You should have the same, please. This early return simplifies the code.

Reply:
Done.